library/psa_crypto.c

```diff
@@ -950,7 +950,20 @@ psa_status_t psa_destroy_key( psa_key_handle_t handle )
 #if defined(MBEDTLS_PSA_CRYPTO_SE_C)
     driver = psa_get_se_driver_entry( slot->lifetime );
     if( driver != NULL )
+    {
+        psa_crypto_prepare_transaction( PSA_CRYPTO_TRANSACTION_DESTROY_KEY );
+        psa_crypto_transaction.key.lifetime = slot->lifetime;
+        psa_crypto_transaction.key.slot = slot->data.se.slot_number;
+        psa_crypto_transaction.key.id = slot->persistent_storage_id;
+        status = psa_crypto_save_transaction( );
+        if( status != PSA_SUCCESS )
+        {
+            /* TOnogrepDO: destroy what can be destroyed anyway */
```

Comment:
`psa_crypto_stop_transaction( );`?

Reply:
Good question. In practice this currently wouldn't make any difference: if `save` failed, the file wasn't created. So the only thing `stop` would do is to zero out the transaction object in memory, which is good hygiene but wouldn't affect the behavior of the code. Should `save` do this on failure? It's the right thing in all cases currently because `save` is only ever used once per transaction. But if `save` was used in the middle of a transaction, the caller might need to read back some of that information on a failure.

Reply:
I went for your suggestion, which is the one that's conceptually simplest.